ros2bag/test/test_play_qos_profiles.py

```diff
@@ -82,3 +82,15 @@ def test_qos_incomplete(self):
         expected_string_regex = re.compile(ERROR_STRING)
         matches = expected_string_regex.search(bag_command.output)
         assert not matches, print('ros2bag CLI did not produce the expected output')
+        
+    def test_qos_negative(self):
```

Comment:
This is a very heavy test and doesn't hit a case that i think right now is incorrect. I think this would be better structured as a unit test on the `interpret_dict_as_qos_profile` function, passing in a dict of values rather than going through launch and subprocess and the filesystem. 

You should have at least 4 separate tests with only a single invalid value in each. I believe the duration values are not being checked correctly right now, since they go through `dict_to_duration`, as they are not `_VALUE_KEYS`. Adding this extra lighter weight unit test should help you catch this error.